ocean_provider/utils/datatoken.py

```diff
@@ -134,13 +129,19 @@ def verify_order_tx(
             "Provider was not able to check the signed message in ProviderFees event\n"
         )
 
-    # check duration
+    timestamp_now = datetime.utcnow().timestamp()
+    # check validUntil
     if provider_fee_order_log.args.validUntil > 0 and not allow_expired_provider_fees:
-        timestamp_now = datetime.utcnow().timestamp()
-        if provider_fee_order_log.args.validUntil < timestamp_now:
-            raise AssertionError(
-                "Validity in transaction exceeds current UTC timestamp"
-            )
+        if timestamp_now >= provider_fee_order_log.args.validUntil:
+            # expired validUntil.  let's add the difference between provider_data["timestamp"](time of initializeCompute) and transaction timestamp.
+            # Also add 90 secs as buffer, since block.timestamp can be manipulated
+            block = web3.eth.get_block(tx_receipt.blockHash, False)
+            if provider_initialize_timestamp > 0 and (
+                timestamp_now
+                >= provider_fee_order_log.args.validUntil
+                + (block.timestamp - provider_initialize_timestamp + 90)
```

Comment:
what does the 90 stand for? is it c2d buffer?

Reply:
yes. let's keep it hardcoded until we need to change it